packages/x-data-grid-pro/src/tests/dataSourceTreeData.DataGridPro.test.tsx

```diff
@@ -129,17 +129,18 @@ describe('<DataGridPro /> - Data source tree data', () => {
   });
 
   it('should fetch nested data when clicking on a dropdown', async () => {
-    render(<TestDataSource />);
+    const { user } = render(<TestDataSource />);
 
+    expect(fetchRowsSpy.callCount).to.equal(1);
     await waitFor(() => {
-      expect(fetchRowsSpy.callCount).to.equal(1);
+      expect(Object.keys(apiRef.current.state.rows.tree).length).to.equal(10 + 1);
+    });
+
+    const cell11 = getCell(0, 0);
+    await act(() => {
+      user.click(within(cell11).getByRole('button'));
     });
```

Comment:
Was this strange approach needed? 🤔 
`user` events are async, having this should be enough
```diff
-await act(() => {
-  user.click(within(cell11).getByRole('button'));
-});
+await user.click(within(cell11).getByRole('button'));
```

Reply:
I wrapped the `fireEvent` that was there and updated to this, but the `act` remained 😅

Updated.